### PR TITLE
Modify datetime picker for scheduling articles

### LINKED ISSUE #18075: Improve the datetime picker for scheduling posts
Scheduling articles was implemented in https://github.com/forem/forem/pull/17939 and can be enabled via `FeatureFlag.enable(:schedule_articles)`.
 
Currently, there is a datetime-local field, but this solution is problematic because the user has to fill the date and time fully, including seconds, otherwise the value won't be saved, which is misleading. We can't pre-populate the field because, in this case, `published_at` would be different from the actual current time, when the user publishes the article w/o scheduling.

[A full discussion about the datetime picker](https://github.com/forem/forem/pull/17939#discussion_r901489598)

One possible solution is to build a custom datetime picker based on `react-dates`. We have already built `DateRangePicker` based on `react-dates`.

Another possible solution is use separate [date](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date) and [time](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time) inputs instead of one `datetime-local`. In the `time` input seconds are optional, not like in our `datetime-local` input. This solution should be simpler than the first one.

Current post options with the datetime picker:
изображение

The logic of saving `published_at` should remain the same.

### PR DESCRIPTION
**Pull request by lightalloy**
## What type of PR is this? (check all applicable)
- [x] Feature

## Description
Replace the datetimepicker in the post options with separate date and time pickers to solve problems described in #18075

## Related Tickets & Documents
- Closes #18075 

## QA Instructions, Screenshots, Recordings
- enable the feature flag: `FeatureFlag.enable(:schedule_articles)`
- create a scheduled article (select date and time in the post options), `published_at` should be saved
  If you choose only date, `published_at` should be saved with the time `00:00`. Choosing only time won't take effect.
- edit an existing scheduled article, it should be possible to set a different `published_at`

изображение

## Added/updated tests?

- [x] Yes
- [x] I need help with writing tests
It would be nice to add at least basic front-end tests. I tried adding a spec to `publishingFlows/postOptions.spec.js`.
To enable a feature flag I added `cy.enableFeatureFlag('schedule_articles');` to `beforeEach()` after the `cy.testSetup();` but it took no effect.

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [x] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

### CODE REVIEW

**Review by Ridhwana — APPROVED** [review]

File: app/javascript/article-form/components/Options.jsx [hunk 1]
```diff
@@ -92,20 +76,31 @@ export const Options = ({
       </div>
     );
   }
+
   if (schedulingEnabled && !readonlyPublishedAt) {
+    const currentDate = moment().format('YYYY-MM-DD');
     publishedAtField = (
       <div className="crayons-field mb-6">
         <label htmlFor="publishedAt" className="crayons-field__label">
```
Comment (aitchiss) [inline app/javascript/article-form/components/Options.jsx:84]:
```suggestion
        <label className="crayons-field__label">
```

we no longer have a `publishedAt` input, and I think we need individual aria-labels anyway

File: app/javascript/article-form/components/Options.jsx [hunk 2]
```diff
@@ -92,20 +76,31 @@ export const Options = ({
...
           Schedule Publication
         </label>
         <input
-          type="datetime-local"
-          min={minPublishedAt}
-          value={localPublishedAt} // "2022-04-28T15:00:00"
+          type="date"
```
Comment (aitchiss) [inline app/javascript/article-form/components/Options.jsx:88]:
```suggestion
          aria-label="Schedule publication date"
          type="date"
```

File: app/javascript/article-form/components/Options.jsx [hunk 3]
```diff
@@ -92,20 +76,31 @@ export const Options = ({
...
+          min={currentDate}
+          value={publishedAtDate} // ""
+          className="crayons-textfield"
+          name="publishedAtDate"
+          onChange={onConfigChange}
+          id="publishedAtDate"
+          placeholder="..."
+        />
+        <input
+          type="time"
```
Comment (aitchiss) [inline app/javascript/article-form/components/Options.jsx:98]:
```suggestion
          aria-label="Schedule publication time"
          type="time"
```

**Review by aitchiss — COMMENTED** [review]
Left a v small a11y tweak 😄 

Nice work finding a simple(ish) solution for this one!

**Pull request merged**